fili-core/src/main/java/com/yahoo/bard/webservice/table/resolver/DataSourceConstraint.java

```diff
@@ -80,4 +104,40 @@ public DataSourceConstraint(DataApiRequest dataApiRequest, DruidAggregationQuery
     public Map<Dimension, Set<ApiFilter>> getApiFilters() {
         return apiFilters;
     }
+
+    /**
+     * Create a new <tt>DataSourceConstraint</tt> instance with a new subset of table columns.
+     * <p>
+     * The new set of table columns will be an intersection between old columns and
+     * a user provided set of columns
+     *
+     * @param other the user provided set of columns
+     *
+     * @return The new <tt>DataSourceConstraint</tt> instance with a new subset of table columns
+     */
+    public DataSourceConstraint withIntersectedColumnNames(Set<String> other) {
+        return new DataSourceConstraint(
+                intersectionBetween(getRequestDimensions(), other),
+                intersectionBetween(getFilterDimensions(), other),
+                intersectionBetween(getMetricDimensions(), other),
+                metricNames.stream()
```

Comment:
Just need the `metricNames` to intersect `other` not the dimensions here, change the name to include `metricIntersection`, leave the dimensions the same.